ledger-service/http-json/src/main/scala/com/digitalasset/http/Endpoints.scala

```diff
@@ -66,64 +67,102 @@ class Endpoints(
   import Uri.Path._
 
+  // Parenthesis in the case matches below are required because otherwise scalafmt breaks.
+  //noinspection ScalaUnnecessaryParentheses
   def all(implicit
       lc: LoggingContextOf[InstanceUUID],
       metrics: Metrics,
   ): PartialFunction[HttpRequest, Http.IncomingConnection => Future[HttpResponse]] = {
-    val dispatch: PartialFunction[HttpRequest, LoggingContextOf[
-      InstanceUUID with RequestID
-    ] => Future[HttpResponse]] = {
-      // Parenthesis are required because otherwise scalafmt breaks.
-      case req @ HttpRequest(POST, Uri.Path("/v1/create"), _, _, _) =>
-        (implicit lc => httpResponse(create(req)))
-      case req @ HttpRequest(POST, Uri.Path("/v1/exercise"), _, _, _) =>
-        (implicit lc => httpResponse(exercise(req)))
-      case req @ HttpRequest(POST, Uri.Path("/v1/create-and-exercise"), _, _, _) =>
-        (implicit lc => httpResponse(createAndExercise(req)))
-      case req @ HttpRequest(POST, Uri.Path("/v1/fetch"), _, _, _) =>
-        (implicit lc => httpResponse(fetch(req)))
+    val apiMetrics = metrics.daml.HttpJsonApi
+    type DispatchFun =
+      PartialFunction[HttpRequest, LoggingContextOf[InstanceUUID with RequestID] => Future[
+        HttpResponse
+      ]]
+    def mkDispatchFunWithTimer(timer: Timer)(fun: DispatchFun): DispatchFun =
+      fun andThen (f => lc => Timed.future(timer, f(lc)))
+    val commandDispatch =
+      mkDispatchFunWithTimer(apiMetrics.commandSubmissionTimer) {
+        case req @ HttpRequest(POST, Uri.Path("/v1/create"), _, _, _) =>
+          (implicit lc => httpResponse(create(req)))
+        case req @ HttpRequest(POST, Uri.Path("/v1/exercise"), _, _, _) =>
+          (implicit lc => httpResponse(exercise(req)))
+        case req @ HttpRequest(POST, Uri.Path("/v1/create-and-exercise"), _, _, _) =>
+          (implicit lc => httpResponse(createAndExercise(req)))
+      }
+    val queryDispatch = mkDispatchFunWithTimer(apiMetrics.queryTimer) {
       case req @ HttpRequest(GET, Uri.Path("/v1/query"), _, _, _) =>
         (implicit lc => httpResponse(retrieveAll(req)))
       case req @ HttpRequest(POST, Uri.Path("/v1/query"), _, _, _) =>
         (implicit lc => httpResponse(query(req)))
+    }
+    val fetchDispatch: DispatchFun = {
+      case req @ HttpRequest(POST, Uri.Path("/v1/fetch"), _, _, _) =>
+        (implicit lc => Timed.future(apiMetrics.fetchTimer, httpResponse(fetch(req))))
+    }
+    val getPartyDispatch = mkDispatchFunWithTimer(apiMetrics.getPartyTimer) {
       case req @ HttpRequest(GET, Uri.Path("/v1/parties"), _, _, _) =>
         (implicit lc => httpResponse(allParties(req)))
       case req @ HttpRequest(POST, Uri.Path("/v1/parties"), _, _, _) =>
         (implicit lc => httpResponse(parties(req)))
+    }
+    val allocatePartyDispatch: DispatchFun = {
       case req @ HttpRequest(POST, Uri.Path("/v1/parties/allocate"), _, _, _) =>
-        (implicit lc => httpResponse(allocateParty(req)))
+        (
```

Comment:
All of this weird formatting just because scalafmt needs these braces, otherwise it fails with a parser error -.-

Reply:
I'll take weird formatting over formatting discussions any time. 🙂